Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -5,6 +5,14 @@ import NIO
 import XCTest
 
 open class ExtendedJSONConversionTestCase: BSONTestCase {
+    func testAnyExtJSON() throws {
+        // Success cases
+        expect(try BSON(fromExtJSON: "hello", keyPath: [])).to(equal(BSON.string("hello")))
+        let document = try BSON(fromExtJSON: ["num": ["$numberInt": "5"], "extra": 1], keyPath: [])
+        expect(document.documentValue!["num"]).to(equal(.int32(5)))
```

Comment:
these `!` casts will crash the test runner if they fail, so it's good practice to include a `expect(document.documentValue).toNot(beNil())` before them to have the test fail more naturally.

Reply:
discussed offline: it's fine to use `!` in tests after asserting non-nil